src/compile/scale/range.ts

```diff
@@ -213,7 +213,7 @@ export function defaultRange(
       const rangeMax = sizeRangeMax(mark, xyRangeSteps, config);
       if (isContinuousToDiscrete(scaleType)) {
         // for now 4 is the default value for range cardinality. we might change it later
-        return interpolateRange(rangeMin, rangeMax, 4);
+        return interpolateRange(rangeMin, rangeMax, config.scale.quantileCount);
```

Comment:
1) For quantize, there should be a separate quantizeCount too. 
2) For threshold, you need to read domain length

> If the number of values in the scale’s domain is N, the number of values in the scale’s range must be N+1.

(from https://github.com/d3/d3-scale/blob/master/README.md#threshold_range)

Reply:
Done